### PR TITLE
PM/ECI defect rates in evaluation

### PR DESCRIPTION
**Pull request by MilesHolland**
Adds code to identify and subsequently modify IP/ECI evaluation label columns into defect rates during the aggregation stage. I also added testing for the existing functionality, and moved the similar code for content safety into its own helper function alongside the new one for IP/ECI.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -59,12 +73,74 @@ def _aggregate_metrics(df, evaluators) -> Dict[str, float]:
...
+        if metric_name.endswith("_label") and metric_name.replace("_label", "").lower() in handled_metrics:
+            label_cols.append(col)
+
+    label_df = df[label_cols]
+    defect_rates = {}
+    for col in label_df.columns:
+        defect_rate_name = col.replace("_label", "_defect_rate")
+        col_with_boolean_values = pd.to_numeric(label_df[col], errors="coerce")
+        defect_rates[defect_rate_name] = round(
+            np.sum(col_with_boolean_values >= 0.5) / col_with_boolean_values.count(),
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:105]: That was somewhat arbitrary - I just assumed that NP represents false as 0 and true as 1, and wanted it to just count true values.

File: src/promptflow-evals/tests/evals/e2etests/test_evaluate.py [hunk 2]
```diff
@@ -178,7 +178,9 @@ def test_evaluate_with_content_safety_evaluator(self, project_scope, data_file):
 
         row_result_df = pd.DataFrame(result["rows"])
         metrics = result["metrics"]
+        import pdb
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply src/promptflow-evals/tests/evals/e2etests/test_evaluate.py]: removed

File: src/promptflow-evals/tests/evals/e2etests/test_builtin_evaluators.py [hunk 3]
```diff
@@ -417,3 +416,6 @@ def test_eci_evaluator(self, project_scope, azure_cred):
         assert unrelated_result is not None
         assert not unrelated_result["ECI_label"]
         assert "geometry question" in unrelated_result["ECI_reasoning"]
+        import pdb
+
```
[the remark this answers is not in the document]
Comment (MilesHolland) [reply src/promptflow-evals/tests/evals/e2etests/test_builtin_evaluators.py]: removed

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py]: changed since apparently it'll work on just the row.

**Review by ninghu — DISMISSED** [review]

**Review by luigiw — APPROVED** [review]

**Review by ninghu — APPROVED** [review]

**Pull request merged**